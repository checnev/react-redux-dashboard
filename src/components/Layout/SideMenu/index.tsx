import React from 'react';
import {
  Drawer,
  MenuList,
  MenuItem,
  ListItemIcon,
  Typography,
  Link
} from '@material-ui/core';
import { NavLink } from 'react-router-dom';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { 
  Dashboard,
  ViewList,
  Report,
  Map,
} from '@material-ui/icons';

const useStyles = makeStyles((theme: Theme) => {
  return createStyles({
  drawer: {
    width: 240,
  },
  drawerPaper: {
    width: 240,
  },
  menuList: {
    paddingTop: 0,
    '&:active': {
      outline: 'none',
    },
  },
  menuItem: {
    padding: 0,
  },
  link: {
    display: 'flex',
    flexGrow: 1,
    alignItems: 'center',
    textDecoration: 'none',
    padding: '6px 16px',
    color: theme.palette.text.primary,
    borderLeft: '5px solid transparent',
    transition: theme.transitions.create(['all']),
    '&:hover': {
      textDecoration: 'none',
    }
  },
  activeLink: {
    backgroundColor: theme.palette.type === 'light' ? 'rgba(63, 81, 181, .15)' : 'rgba(51, 51, 51, .15)',
  }
})
});

const menuItems = [
  { icon: <Dashboard />, anchor: 'Главная панель', to: '/' },
  { icon: <ViewList />, anchor: 'Автомобили', to: '/cars' },
  { icon: <Map />, anchor: 'Карта', to: '/maps' },
  { icon: <Report />, anchor: 'Отзывы', to: '/reviews' },
];

interface SideMenuProps {
  isDrawerOpen: boolean;
  closeDrawer: () => void;
}

const SideMenu: React.FC<SideMenuProps> = ({ isDrawerOpen, closeDrawer }: SideMenuProps) => {
  const classes = useStyles();

  return (
    <Drawer
      className={classes.drawer}
      onClose={closeDrawer}
      open={isDrawerOpen}
      classes={{
        paper: classes.drawerPaper,
      }}
    >
      <nav>
        <MenuList>
          {menuItems.map((item) => (
            <MenuItem key={item.to} className={classes.menuItem}>
              <Link
                component={NavLink}
                to={item.to}
                className={classes.link}
                activeClassName={classes.activeLink}
                exact
              >
                <ListItemIcon>{item.icon}</ListItemIcon>
                <Typography variant="inherit">{item.anchor}</Typography>
              </Link>
            </MenuItem>
          ))}
        </MenuList>
      </nav>
    </Drawer>
  );
};

export default SideMenu;
