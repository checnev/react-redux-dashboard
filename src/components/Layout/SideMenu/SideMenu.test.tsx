import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import SideMenu from 'components/Layout/SideMenu';

import { AppBar } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';

let component: any = null;
beforeEach(() => {
  component = document.createElement('div');
  document.body.append(component);
});

afterEach(() => {
  unmountComponentAtNode(component);
  component.remove();
  component = null;
});

const closeDrawer = jest.fn();

describe('SideMenu', () => {
  const setUp = (props: any) => (
    <Router>
      <AppBar>
        <SideMenu {...props} />
      </AppBar>
    </Router>
  );

  it('drawer should be close', () => {
    act(() => {
      render(setUp({ isDrawerOpen: false, closeDrawer }), component);
    });

    expect(document.querySelector('.MuiDrawer-root')).toBeNull();
  });

  it('drawer should be open', () => {
    act(() => {
      render(setUp({ isDrawerOpen: true, closeDrawer }), component);
    });

    expect(
      pretty(document.querySelector('.MuiDrawer-root')!.innerHTML)
    ).toMatchSnapshot();
  });
});
