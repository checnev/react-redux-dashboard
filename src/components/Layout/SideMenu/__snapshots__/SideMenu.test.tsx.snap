// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu drawer should be open 1`] = `
"<div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div>
<div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div>
<div class=\\"MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-8 MuiDrawer-paperAnchorLeft MuiPaper-elevation16\\" tabindex=\\"-1\\">
  <nav>
    <ul class=\\"MuiList-root MuiList-padding\\" role=\\"menu\\" tabindex=\\"-1\\">
      <li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-10 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"menuitem\\" aria-disabled=\\"false\\"><a aria-current=\\"page\\" class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-11 MuiTypography-colorPrimary makeStyles-activeLink-12\\" href=\\"/\\">
          <div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
              <path d=\\"M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z\\"></path>
            </svg></div><span class=\\"MuiTypography-root\\">Главная панель</span>
        </a><span class=\\"MuiTouchRipple-root\\"></span></li>
      <li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-10 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-11 MuiTypography-colorPrimary\\" href=\\"/cars\\">
          <div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
              <path d=\\"M4 14h4v-4H4v4zm0 5h4v-4H4v4zM4 9h4V5H4v4zm5 5h12v-4H9v4zm0 5h12v-4H9v4zM9 5v4h12V5H9z\\"></path>
            </svg></div><span class=\\"MuiTypography-root\\">Автомобили</span>
        </a><span class=\\"MuiTouchRipple-root\\"></span></li>
      <li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-10 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-11 MuiTypography-colorPrimary\\" href=\\"/maps\\">
          <div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
              <path d=\\"M20.5 3l-.16.03L15 5.1 9 3 3.36 4.9c-.21.07-.36.25-.36.48V20.5c0 .28.22.5.5.5l.16-.03L9 18.9l6 2.1 5.64-1.9c.21-.07.36-.25.36-.48V3.5c0-.28-.22-.5-.5-.5zM15 19l-6-2.11V5l6 2.11V19z\\"></path>
            </svg></div><span class=\\"MuiTypography-root\\">Карта</span>
        </a><span class=\\"MuiTouchRipple-root\\"></span></li>
      <li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-10 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-11 MuiTypography-colorPrimary\\" href=\\"/reviews\\">
          <div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
              <path d=\\"M15.73 3H8.27L3 8.27v7.46L8.27 21h7.46L21 15.73V8.27L15.73 3zM12 17.3c-.72 0-1.3-.58-1.3-1.3 0-.72.58-1.3 1.3-1.3.72 0 1.3.58 1.3 1.3 0 .72-.58 1.3-1.3 1.3zm1-4.3h-2V7h2v6z\\"></path>
            </svg></div><span class=\\"MuiTypography-root\\">Отзывы</span>
        </a><span class=\\"MuiTouchRipple-root\\"></span></li>
    </ul>
  </nav>
</div>
<div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div>"
`;
